python-oop/static-methods-and-class-methods.md

```diff
@@ -9,25 +9,260 @@
+
+```python
+class ExampleClass:
+
+    @staticmethod
+    def example_method():
+        print("I'm inside the static method, example_method!")
+```
+
+Once a static method is defined, we can invoke it using dot notation, typically from the class itself.
+
+```python
+ExampleClass.example_method()
+```
+
+This code produces the console output:
+
+```
+I'm inside the static method, example_method!
+```
+
+Any code that can reference the class can call a static method. This means that we can call a static method from another static method, class method, instance method, or even inside another class.
+
+### Static Methods: Syntax Details
+
+In Python, we actually _can_ call static methods from an object instance, not just a class. However, because we don't pass in `self`, we don't have a reference to `self`. Static methods do not depend on an instance.
+
+Observe this code, which shows that we _can_ call static methods from an object instance, but we can't access `self`.
+
+```python
+class ExampleClass:
+
+    def __init__(self, name):
+        self.name = name
+
+    @staticmethod
+    def example_method():
+        print("I'm inside the static method, example_method!")
+        print(f"I'm accessing an instance variable, {self.name}")
+```
+
+```python
+example_instance = ExampleClass("Hello, World!")
+
+example_instance.example_method()
+```
+
+This code produces the console output:
+
+```
+I'm inside the static method, example_method!
+Traceback (most recent call last):
+  File "main.py", line 13, in <module>
+    example_instance.example_method()
+  File "main.py", line 9, in example_method
+    print(f"I'm accessing an instance variable, {self.name}")
+NameError: name 'self' is not defined
+```
+
+We should observe from the output that:
+
+- The code _did_ successfully begin the static method, and it _did_ print `I'm inside the static method, example_method!`
+- The code threw a runtime error `NameError`, because we are unable to access `self`
+
 ## Static Methods: Application
 
+Static methods are often utility functions. "Utility functions" is an informal term to describe a function responsible for doing common, routine tasks.
+
+Static methods are great for:
+
+- returning fixed values not associated with an instance
+- doing operations that aren't associated with an instance
+
 ## Class Methods are Methods with Class State
 
+Class methods are methods that are only called from a class, not an instance.
+
+Class methods have access to class variables. However, they do _not_ have access to any instance variables.
+
 ## Class Methods: Syntax
 
+To define a class method, we:
+
+- use the `@classmethod` decorator that the Python library provides
+- do not pass in `self` as a parameter
+- pass in `cls` as a parameter
+
+Instead of passing in `self`, the variable that represents an instance itself, in class methods, we pass in `cls`, a variable that represents the class itself.
+
+`cls` is the conventional name for this parameter.
+
```

Comment:
I think stating this one in the bullets is fine since the example comes right away.
```suggestion
```